sockeye/inference.py

```diff
@@ -1532,11 +1520,14 @@ def _beam_search(self,
                                                                               attention_scores)
 
             # (5) Normalize the scores of newly finished hypotheses. Note that after this until the
-            # next call to topk(), hypotheses may not be in sorted order.
-            finished, scores_accumulated = self._normalize_finished.forward(best_word_indices,
-                                                                            finished,
-                                                                            scores_accumulated,
-                                                                            lengths)
+            # next call to topk(), hypotheses may not be in sorted order. Don't update `finished` yet;
+            # we need to include newly-finished (on this time-step) items for normalization and pruning,
+            # but we need the old values of `finished` to update lengths first, using the old values
+            finished, scores_accumulated, lengths = self._update_finished.forward(best_word_indices,
```

Comment:
I don't get the comment about not updating `finished` yet. From the perspective in `_beam_search`, `finished` is returned from the block and is in fact different than the `finished` passed in as an argument. Moreover, the block is called `NormalizeAndUpdateFinished`.

Reply:
This comment applied when we were going to update the length by one, to include `<s>`. Since we decided not to, this comment no longer applies. Deleting.